src/lib.rs

```diff
@@ -427,6 +427,30 @@ impl<L: AnalysisLoader> AnalysisHost<L> {
         result
     }
 
+    pub fn borrow_info(&self, span: &Span) -> AResult<BorrowData> {
+        self.with_analysis(|a|
+            a.def_id_for_span(span)
+             .and_then(|id| a.for_each_crate(|c| {
+                trace!("Searching in crate `{}`", c.name);
+                for (_, b) in c.per_fn_borrows.iter() {
+                    // If we find a `BorrowData` where there's a matching scope, then filter out
+                    // out matching items.
+                    if b.scopes.iter().any(|a| a.ref_id == id) {
```

Comment:
Hmm, that is not something that is easy to do (at the moment), I think. We could save a 'map' of spans to function ids, then do some kind of sub-span check to get the function id. I think that should be relatively easy if the 'map' is the correct data structure (maybe just an ordered Vec is fine, but building it would be easier if it were some kind of tree) and I think the memory overhead wouldn't be too high.